packages/synthetics-sdk-api/src/cloud_region_resolver.ts

```diff
@@ -0,0 +1,48 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import axios, { AxiosResponse } from 'axios';
+
+/**
+ *
+ * @public
+ *
+ * Retrieves the region in which the current Google Cloud Function (v2) is
+ * executing by querying the metadata server.
+ */
+export async function getExecutionRegion(): Promise<string | null> {
```

Comment:
This isn't stored somewhere within the environment variables of the GCF?

Reply:
Based on YAQs, quick convo with engineer, and a quick search I believe this is the only way.